File: frontend/src/types/quality.ts
import { IPeriod } from './period'

export interface IQuality {
	id: string
	date: number
	// type: 'final' | 'fixable'
	product: string
	title: string
	count: number
	number: number
	time: number
	// 	amount: number
	// 	percent: number
	// 	cost: number
}

export interface IQualityDTO {
	id?: string
	date?: number
	// type?: 'final' | 'fixable'
	product?: string
	title: string | null
	count: number | null
	number: number | null
	time: number | null
	// amount: number | null
	// percent: number | null
	// cost: number | null
}

export interface IGetQuality {
	product: string
	period: IPeriod
}

File: backend/internal/models/quality.go
package models

// type Quality struct {
// 	Id      string  `json:"id" db:"id"`
// 	Date    int64   `json:"date" db:"date"`
// 	Type    string  `json:"type" db:"type"`
// 	Product string  `json:"product" db:"product"`
// 	Title   string  `json:"title" db:"title"`
// 	Number  int     `json:"number" db:"number"`
// 	Amount  int     `json:"amount" db:"amount"`
// 	Percent int     `json:"percent" db:"percent"`
// 	Cost    float64 `json:"cost" db:"cost"`
// }

type Quality struct {
	Id      string  `json:"id" db:"id"`
	Date    int64   `json:"date" db:"date"`
	Product string  `json:"product" db:"product"`
	Title   string  `json:"title" db:"title"`
	Count   int     `json:"count" db:"count"`
	Number  int     `json:"number" db:"number"`
	Time    float64 `json:"time" db:"time"`
}

type GetQualityDTO struct {
	Product string  `json:"product"`
	Period  *Period `json:"period"`
}
